apps/authentication-portal/src/main/webapp/basicauth.jsp

```diff
@@ -292,10 +297,16 @@
             urlEncodedURL = URLEncoder.encode(urlWithoutEncoding, UTF_8);
             urlParameters = prmstr;
 
-            identityMgtEndpointContext =
-                    application.getInitParameter("IdentityManagementEndpointContextURL");
+            identityMgtEndpointContext = application.getInitParameter("IdentityManagementEndpointContextURL");
             if (StringUtils.isBlank(identityMgtEndpointContext)) {
-                identityMgtEndpointContext = getServerURL("/accountrecoveryendpoint", true, true);
+                try {
+                    identityMgtEndpointContext = ServiceURLBuilder.create().addPath(ACCOUNT_RECOVERY_ENDPOINT).build()
+                            .getAbsolutePublicURL();
+                } catch (URLBuilderException e) {
+                    String redirectURL = "error.do?" + STATUS + "=" + CONFIGURATION_ERROR + "&" + STATUS_MSG + "="
+                            + ERROR_WHILE_BUILDING_THE_TENANT_QUALIFIED_URL;
+                    response.sendRedirect(redirectURL);
```

Comment:
Rather than using a redirect let's forward internally.
You can follow the approach here.
https://github.com/wso2/identity-apps/blob/master/apps/recovery-portal/src/main/webapp/self-registration-with-verification-confirm.jsp#L70

Reply:
Updated the PR with the changes.